src/main/java/org/apache/maven/plugins/checkstyle/CheckstyleViolationCheckMojo.java

```diff
@@ -624,51 +625,104 @@ private int countViolations( XmlPullParser xpp )
+                category
+            );
+            if ( column != null )
+            {
+                violation.setColumn( column );
+            }
 
-                String source = xpp.getAttributeValue( "", "source" );
+            violations.add( violation );
+        }
 
-                if ( ignore( ignores, source ) )
-                {
-                    ignoreCount++;
-                }
-                else
-                {
-                    count++;
+        return Collections.unmodifiableList( violations );
+    }
 
-                    if ( logViolationsToConsole )
-                    {
-                        String line = xpp.getAttributeValue( "", "line" );
-                        String column = xpp.getAttributeValue( "", "column" );
-                        String message = xpp.getAttributeValue( "", "message" );
-                        String rule = RuleUtil.getName( source );
-                        String category = RuleUtil.getCategory( source );
-
-                        log( severity, file + ":[" + line + ( ( column == null ) ? "" : ( ',' + column ) ) + "] ("
-                            + category + ") " + rule + ": " + message );
-                    }
-                }
+    private int countViolations( List<Violation> violations )
+    {
+        List<RuleUtil.Matcher> ignores = violationIgnore == null ? Collections.<RuleUtil.Matcher>emptyList()
+            : RuleUtil.parseMatchers( violationIgnore.split( "," ) );
+
+        int ignored = 0;
+
+        List<Violation> actualViolations = new ArrayList<>();
+
+        for ( Violation violation : violations )
+        {
+            if ( ! isViolation( violation.getSeverity() ) )
+            {
+                continue;
+            }
+
+            if ( ignore( ignores, violation.getSource() ) )
+            {
+                ignored++;
+                continue;
             }
+
+            actualViolations.add( violation );
         }
 
-        if ( ignoreCount > 0 )
+        final int count = actualViolations.size();
+
+        if ( ignored > 0 )
         {
-            getLog().info( "Ignored " + ignoreCount + " error" + ( ( ignoreCount > 1 ) ? "s" : "" ) + ", " + count
-                               + " violation" + ( ( count > 1 ) ? "s" : "" ) + " remaining." );
+            getLog().info( "Ignored " + ignored + " error" + ( ( ignored > 1L ) ? "s" : "" ) + ", " + count
+                + " violation" + ( ( count > 1 ) ? "s" : "" ) + " remaining." );
         }
 
         return count;
     }
 
+    private void printViolations( List<Violation> violations )
+    {
+        if ( ! logViolationsToConsole )
+        {
+            return;
+        }
+
+        List<RuleUtil.Matcher> ignores = violationIgnore == null ? Collections.<RuleUtil.Matcher>emptyList()
+            : RuleUtil.parseMatchers( violationIgnore.split( "," ) );
+
+        violations.stream()
+            .filter( violation -> isViolation( violation.getSeverity() ) )
```

Comment:
This is weird. We have a list of violations, but some of them aren't violations?! perhaps isViolation could be renamed something like isImportant or isSevereEnoughToCount (not quite right, but you get the idea)

Reply:
How big do you want this commit to be? Originally I only wanted to create a `Violation.java` class. But now you are asking me to rename every existing variable and method while I am at it? Is this really how a review should work?

Sorry if this comment sounds intense, no offense intended. I am just wondering what else I should do whenever I submit a PR. Always check for errors of other programmers? Why have you not mentioned this to the guy who created this method in the first place?

I really suggest to open a new issue for such things. I really do not want to rewrite the entire mojo here. The reason I started this PR was more like: Hmm, the countViolations() method does multiple things besides counting (printing etc.). Let's make that more maintainable.

But yes, I thing the intention of the `isViolation()` method is more like `isToBeMentioned` or something.

Please kindly open another issue for this and let me not touch this method, because I am still using it. I am happy to rename it for another JIRA issue.

Reply:
Please do open an issue for this. This is new code you've written and it's not clear as is. It's not surprising that introducing a new class might reveal other problems that need to be fixed at the same time.

Reply:
https://issues.apache.org/jira/browse/MCHECKSTYLE-386